gps_exif_bot2.py

```diff
@@ -1,523 +1,309 @@
-    text = ""
-
-    if page.exists() :
-      # follow redirects to the actual image page
-      while True :
-        try :
-          text = page.get()
-          break
-        except pywikibot.IsRedirectPage :
-          page = page.getRedirectTarget()
-        except :
-          print "Could not get page " + name
-	  taglist[name] = False
-          text = False
-          break
-
-    else :
-      continue
-	
-    if text == False :
-      continue
+connection = MySQLdb.connect(
+    host="commonswiki.labsdb",
+    db="commonswiki_p",
+    read_default_file="~/replica.my.cnf")
+cursor = connection.cursor()
+cursor.execute("""\
+-- https://quarry.wmflabs.org/query/26631
+SELECT page_title
+    FROM page
+INNER JOIN image
+    ON img_name = page_title
+LEFT JOIN templatelinks
+    ON tl_from = page_id
+    AND tl_namespace = 10
+    AND tl_title IN ('Location', 'Location_dec')
+WHERE page_namespace = 6
+    AND (
+        img_metadata LIKE '%"GPSAltitude"%' OR
+        img_metadata LIKE '%"GPSLatitudeRef"%'
+    )
+
+    -- select specific images (debug)
+    -- AND page_title IN (
+    --    'Chur_de_la_basilique_Saint-Sauveur,_Dinan,_France.jpg')
+
+    -- go over all uploads of a specific user (manual)
+    -- AND img_user_text = "Niteshift"
+
+    -- contains no relevant template
+    AND tl_from IS NULL
+
+    -- look at images of the last two days
+    AND img_timestamp > DATE_FORMAT(NOW() - INTERVAL 2 DAY, '%Y%m%d%H%i%S')
+UNION
+SELECT page_title
+    FROM page, templatelinks, image
+WHERE tl_namespace = 10
+    AND tl_title = 'GPS_EXIF'
+    AND page_id = tl_from
+    AND page_namespace = 6
+    AND page_title = img_name
+;""")
+
+data = cursor.fetchall()
+cursor.close()
+connection.close()
+
+
+def num_str(value):
+    return ('%.5f' % value).rstrip('0').rstrip('.')
+
+
+def deg_min_sec(*args):
+    deg = 0.0
+    for i, element in enumerate(args):
+        deg += float(element) / 60**i
+
+    return deg, '|'.join(map(num_str, args))
+
+
+def extract_exif_latlong(metadata, latlong, posdir, negdir):
+    if 'Exif.GPSInfo.GPSLongitude' in metadata.exif_keys:
+        dec, params = deg_min_sec(
+            *metadata['Exif.GPSInfo.GPS' + latlong].value)
+        ref = metadata['Exif.GPSInfo.GPS' + latlong + 'Ref'].value
+    else:
+        exif = metadata['Xmp.exif.GPS' + latlong].value
+        dec, params = deg_min_sec(exif.degrees, exif.minutes, exif.seconds)
+        ref = metadata['Xmp.exif.GPS' + latlong].value.direction
+
+    if ref == negdir:
+        dec = -dec
+    elif ref != posdir:
+        raise RuntimeError('Broken %s ref: %s!' % (latlong.lower(), ref))
+
+    return dec, '|' + params + '|' + ref
+
+
+def already_processed(name, page):
+    pywikibot.output("HMM, %s looks already processed" % name)
+    page.save("removed gps exif request template")
+
+
+def process_image(name):
+    page = pywikibot.FilePage(site, name.decode('utf-8'))
+    page.save = lambda *args, **kwargs: None
```

Comment:
Oops, I made the script run in 'simulation' with this line. Could you remove it?